tests/test_wrappers.py

```diff
@@ -551,6 +559,48 @@ def test_etag_response_mixin():
     assert response.content_length == 999
 
 
+def test_etag_response_412():
+    response = wrappers.Response('Hello World')
+    assert response.get_etag() == (None, None)
+    response.add_etag()
+    assert response.get_etag() == ('b10a8db164e0754105b7a99be72e3fe5', False)
+    assert not response.cache_control
+    response.cache_control.must_revalidate = True
+    response.cache_control.max_age = 60
+    response.headers['Content-Length'] = len(response.get_data())
+    assert response.headers['Cache-Control'] in ('must-revalidate, max-age=60',
+                                                 'max-age=60, must-revalidate')
+
+    assert 'date' not in response.headers
+    env = create_environ()
+    env.update({
+        'REQUEST_METHOD':       'GET',
```

Comment:
what's the reason for this rather weird spacing?

Reply:
done